gitbook-docs/market/user-guides/proof-producer.md

```diff
@@ -1,82 +1,88 @@
+   ./bin/proof-generator/proof-generator --proof_out=<output file> \
+   --circuit_input=<statement from Proof Market> --public_input=<public input from Proof Market>
 # Or using the multithreaded version
-./bin/proof-generator/proof-generator-mt --proof_out=<output file> --circuit_input=<statement from Proof Market> --public_input=<public input from Proof Market> --smp=<number of threads> ----shard0-mem-scale=<scale>
-
+   ./bin/proof-generator/proof-generator-mt --proof_out=<output file> \
+   --circuit_input=<statement from Proof Market> --public_input=<public input from Proof Market> \
+   --smp=<number of threads> ----shard0-mem-scale=<scale>
 ```
 
-For multithreaded versions, the following flags warrant discussion as the computation of proof requires temporary space. This is always allocated to `core0`/`shard0.`
+Proof computation requires temporary space, always allocated to `core0`/`shard0`.
+The following flags require attention for multithreaded use:
 
-* `smp` : Number of threads. This is ideally the number of CPU cores on the system. ex: If you have a 16 core CPU &  16 GB RAM & set `smp` to 16. Each core gets access to 1 GB of RAM. Thus `core0`/`shard0` will have access to 1 GB of RAM.
-* `shard0-mem-scale`: Weighted parameter (`weight`) to reserve memory for `shard0.`This is a natural number which allows for `shard0` to have more access to RAM in comparison to others. ex: If you have a 16 core CPU &  16 GB RAM & set `smp` to 8 and set `shard0-mem-scale` to 9.
+* `smp` - number of threads.
+  This is ideally the number of CPU cores on the system.
+  If you have a 16-core CPU & 16 GB RAM, you can set `smp` to 16.
+  Each core gets access to 1 GB of RAM, so `core0`/`shard0` will have access to 1 GB of RAM.
+* `shard0-mem-scale` - weighted parameter (`weight`) to reserve memory for `shard0`.
+  This is a natural number that allows for `shard0` to have more access to RAM in comparison to others.
+  If you have a 16-core CPU & 16 GB RAM, you can set `smp` to 8 and `shard0-mem-scale` to 9.
 
-&#x20;We first compute  `ram_per_shard` variable as follows:
+We first compute `ram_per_shard` variable as follows:
 
-```
+```bash
 ram_per_shard = TOTAL_RAM / (smp + shard0-mem-scale -1)
               = 16 / (8 + 8)
               = 1
 ```
 
-This equates to the following:&#x20;
+Which equates to the following:
 
 * `shard0` RAM =`shard0-mem-scale` \* `ram_per_shard` = 9 GB
-* `shard1` .... `shard7` = `ram_per_shard` = 1 GB/per core= 7 GB
+* `shard1` .... `shard7` = `ram_per_shard` = 1 GB/per core = 7 GB
 
-These two variables need to be tuned per the architecture/circuit for which the proof is generated.
+You should tune these two variables for the architecture/circuit for which the proof is generated.
 
+## Submit a proof
 
+The proof producer can now submit the proof to the marketplace, where they will get the reward
+if the proof is verified.
```

Comment:
If it is verified by whom? I guess it's Proof Market, but readers can understand it in a different way.

And we can give a link to the market.md page here, to the part about rewards and punishments.

Reply:
Fixed